src/main/scala/fairysf/format/FEN.scala

```diff
@@ -28,6 +28,15 @@ final case class FEN(value: String) extends AnyVal {
     }
   }
 
+  def invertPlayer: Option[FEN] =
+    player.map { p =>
+      val parts = value.split(' ')
+      // Defensive: ensure there's a part to update
+      if (parts.length > 1) FEN(parts.updated(1, (!p).letter.toString).mkString(" "))
```

Comment:
I don't feel particularly strongly about it, but my preference would be to not bother with the defensive programming, leave a comment, and add some unit tests for this method.